### PR TITLE
Cartocss styles editor

### LINKED ISSUE #7445: Add CartoCSS editor to style form view
Branch: `cartocss-styles-editor`.

### PR DESCRIPTION
**Pull request by nobuti**
cartocss

This PR closes #7445. Minor styles issues need to be solved yet.

@xavijam @piensaenpixel

### CODE REVIEW

**Comment by javisantana** [comment]
@xavijam with this we could do the custom_style thing and integrate it in the styles manager

**Comment by xavijam** [comment]
Yes! Looks amazing man :).

**Comment by piensaenpixel** [comment]
@nobuti we need change two small things. I can help you tomorrow.
- icon in the header
- color link in the header

**Comment by xavijam** [comment]
Check conflicts @nobuti! :)

**Comment by nobuti** [comment]
@xavijam all under control for now, I'd rather to resolve all of them once we finished the work here. There are a couple things to fix yet.

File: lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-view.js [hunk 1]
```diff
@@ -74,10 +75,16 @@ module.exports = cdb.core.View.extend({
...
-    var tabPaneContentView = model.get('createContentView').call(model);
+    var tabPaneContentView = model.get(this._createContentKey).call(model);
     this.addView(tabPaneContentView);
     this.$('.js-content').append(tabPaneContentView.render().$el);
 
     return tabPaneContentView;
+  },
+
+  changeStyleMenu: function (m) {
+    (m.isEditing() === true)
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-view.js]: this kind of expression can be shortened to:

```js
this.$('.js-theme').toggleClass('is-dark', m.isEditing());
```

File: lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js [hunk 2]
```diff
@@ -0,0 +1,343 @@
+/*
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js]: I know it is a pain in the neck, but I'll rename `CodeMirror` files  from code-mirror to codemirror or viceversa. As you wish. Just for consistency.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js]: BTW, @javisantana we need to add turbo-carto definition, and maybe some other changes. Does it make sense to start using new/last css definition from CodeMirror and apply our changes?

File: lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+          n -= state.stack[state.stack.length - 1] === 'rule' ? 2 : 1;
+        }
+        return state.baseIndent + n * indentUnit;
+      },
+
+      electricChars: '}'
+    };
+  });
+
+  CodeMirror.defineMIME('text/x-carto', 'carto');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js]: cartocss

File: lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    'orchid', 'palegoldenrod', 'palegreen', 'paleturquoise', 'palevioletred',
+    'papayawhip', 'peachpuff', 'peru', 'pink', 'plum', 'powderblue',
+    'purple', 'red', 'rosybrown', 'royalblue', 'saddlebrown', 'salmon',
+    'sandybrown', 'seagreen', 'seashell', 'sienna', 'silver', 'skyblue',
+    'slateblue', 'slategray', 'snow', 'springgreen', 'steelblue', 'tan',
+    'teal', 'thistle', 'tomato', 'turquoise', 'violet', 'wheat', 'white',
+    'whitesmoke', 'yellow', 'yellowgreen'
+  ];
+
+  CodeMirror.defineMode('carto', function (config) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js]: cartocss

File: lib/assets/javascripts/cartodb3/editor/components/code-mirror/code-mirror-view.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  },
+
+  _initViews: function () {
+    var self = this;
+    this.editor = CodeMirror.fromTextArea(this.$('.js-editor').get(0), {
+      lineNumbers: true,
+      styleActiveLine: true,
+      matchBrackets: true,
+      theme: 'material',
+      mode: 'carto'
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/code-mirror/code-mirror-view.js]: cartocss please :)

**Comment by xavijam** [comment]
:clap:

File: lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/code-mirror/cartocss.codemirror.js]: Ok, no changes about that. Let's keep with what we already have.

**Comment by xavijam** [comment]
After that, ready to go, I'll take the next part :).

**Comment by xavijam** [comment]
Ready to go buti, travis frontend tests passing. 🍏

**Pull request merged**